src/backend/marsha/core/views.py

```diff
@@ -61,12 +101,16 @@ def get_context_data(self):
                 used as authentication.
 
         """
-        lti = self.get_lti()
+        lti = LTI(self.request, self.kwargs["uuid"])
         try:
-            video = lti.get_or_create_video()
+            resource = self._get_resource(lti)
```

Comment:
you don't need the _get_resource anymore and you also don't need 2 function get_or_create_video and get_or_create_file. Just call:
```python
resource = get_or_create_resource(self.model, lti)
```